conanfile.py

```diff
@@ -160,13 +165,38 @@ def build(self):
         cmake.configure()
         cmake.build()
 
+        if self.options.get_safe("enable_sentry", False):
+            # Upload debug symbols to sentry
+            sentry_project = self.conf.get("user.curaengine:sentry_project", "", check_type=str)
+            sentry_org = self.conf.get("user.curaengine:sentry_org", "", check_type=str)
+            if sentry_project == "" or sentry_org == "":
+                raise ConanInvalidConfiguration("sentry_project or sentry_org is not set")
+            if which("sentry-cli") is None:
+                self.output.warn("sentry-cli is not installed, skipping uploading debug symbols")
+                self.output.warn("sentry-cli is not installed, skipping release creation")
+                return
+
+            if self.settings.os == "Linux":
+                self.output.info("Stripping debug symbols from binary")
+                self.run("objcopy --only-keep-debug --compress-debug-sections=zlib CuraEngine CuraEngine.debug")
+                self.run("objcopy --strip-debug --strip-unneeded CuraEngine")
+                self.run("objcopy --add-gnu-debuglink=CuraEngine.debug CuraEngine")
```

Comment:
I love that... However, I have a bit of a concern because it means the release executable will actually be built using debug mode, and it will not do any code optimization, so execution could be slightly less performant. Am I right ?

Reply:
yes we're now build this as RelWithDebug which is o2. Previous tests didn't indicate a increase in slice time.

But even if it did I would argue that we should find this acceptable since one of our main goals this year is to make the engine more robust and simply said we need that extra information for Sentry. Otherwise we basically only get an event that says crash occurred with no meaningful information what so ever.

It is also not that uncommon and sometimes even considered good practice to release with o2 instead of o3

Reply:
That would be a noticeable drop in speed indeed